src/GafferScene/UDIMQuery.cpp

```diff
@@ -206,8 +207,32 @@ struct InfoDataAccumulator
 		const IECoreScene::MeshPrimitive *meshPrimitive = runTimeCast<const IECoreScene::MeshPrimitive>( object.get() );
 		if( !meshPrimitive ) return true;
 
-		auto uvs = meshPrimitive->variableIndexedView<IECore::V2fVectorData>( m_uvSet,  IECoreScene::PrimitiveVariable::FaceVarying );
-		if( !uvs ) return true;
+		// Kinda funny that with all this C++ gunk around, we still have to resort to using a pointers in the
+		// end because IndexedView doesn't support move semantics
+		auto optUvs = meshPrimitive->variableIndexedView<IECore::V2fVectorData>( m_uvSet,  IECoreScene::PrimitiveVariable::FaceVarying );
+
+		IECoreScene::PrimitiveVariable vertexUvs;
+		std::unique_ptr< IECoreScene::PrimitiveVariable::IndexedView<Imath::V2f> > vertexUvView;
+		IECoreScene::PrimitiveVariable::IndexedView<Imath::V2f> *uvView;
+		if( !optUvs )
+		{
+			// No face varying UVs, check for vertex UVs
+			auto findUvs = meshPrimitive->variables.find( m_uvSet );
+			if( findUvs == meshPrimitive->variables.end() || findUvs->second.interpolation != IECoreScene::PrimitiveVariable::Vertex )
+			{
+				// No vertex UVs either, just bail out
+				return true;
+			}
+			vertexUvs = findUvs->second;
+			MeshAlgo::resamplePrimitiveVariable( meshPrimitive, vertexUvs, IECoreScene::PrimitiveVariable::FaceVarying );
```

Comment:
It looks very much like you're modifying the UVs in place here, but they belong to a const MeshPrimitive which is already in the cache. Am I missing something?

Can we just avoid resampling? In the Wireframe node, I've just used the meshes vertexIds as the indices for vertex UVs, which makes the code the same as for the facevarying case.

Reply:
For what it's worth, [here's](https://github.com/GafferHQ/gaffer/blob/97bca829bc53f00e85bac22cb1710f4bf0601711/src/GafferScene/Wireframe.cpp#L90-L107) how I'm dealing with this in the wireframe node - I've just updated it to cope with all permutations of vertex/varying/facevarying and index/nonindexed.